src/services/fetchProductsFunction.js

```diff
@@ -0,0 +1,12 @@
+const API_BASE_URL = import.meta.env.VITE_BACKEND_URL;
+
+const fetchProductsFunction = (offset, setOffset, filteredCategory, setFilteredCategory, filter) => {
+	fetch(`${API_BASE_URL}/products?filter=${filter}&offset=${offset}&limit=4`)
```

Comment:
no need for `API_BASE_URL`
the api link for front end should be: `/api/v1/products?....whatever....`